src/App.js

```diff
@@ -1,19 +1,37 @@
-import React from 'react'; 
+import React, { useEffect } from 'react'; 
 import AddToDoForm from './addToDoForm';
 import ToDoList from './todolist';
 
-
-
 function App() {
-  const [toDoList, setToDoList]= React.useState([]);
+   //read the list from local storage after load 
+  const [toDoList, setToDoList]= React.useState([]); 
+  const [isLoading, setIsLoading]= React.useState(true);
   const addToDo= (newToDo)=> {
     setToDoList([...toDoList, newToDo])
   }
+  useEffect(()=> {
+    new Promise((resolve, reject)=> setTimeout(()=> resolve({data: {toDoList: []} }), 2000))
+    .then((result)=> {
+      setToDoList(result.data.toDoList);
+      setIsLoading(false);
+    })
+  }, []);
+  //setting list from the input box
+  useEffect(()=> {
+    if (isLoading === false) {
+      localStorage.setItem('savedToDoList', JSON.stringify(toDoList))
+    }
+  }, [toDoList]);
```

Comment:
I've asked about this with the mentors and havne't gotten an answer but: technically, this hook should also depend on `isLoading` b/c it's used in the hook.  It works fine as it is and the lesson said do it this way, so it's fine.  Just thought I'd leave a note.

Reply:
Ah thank you, I was a little confused about that.